Modules/QuestieProfessions.lua

```diff
@@ -124,6 +125,52 @@ function QuestieProfessions:HasProfessionAndSkillLevel(requiredSkill)
     return _HasProfession(profession), _HasSkillLevel(profession, skillLevel)
 end
 
+---@param requiredSpecialization { [1]: number } [1] = professionId
+---@return boolean HasSpecialization
+function QuestieProfessions:HasSpecialization(requiredSpecialization)
+    if not requiredSpecialization then
+        --? We return true here because otherwise we would have to check for nil everywhere
+        return true
+    end
+    for _, value in pairs(QuestieProfessions.professionKeys) do
```

Comment:
You could store `local professionKeys = QuestieProfessions.professionKeys` before the for-loop together with `QuestieProfessions.specializationKeys` to improve readability of the lower block.